### PR TITLE
Implement caching logic for a gems layer

### PR DESCRIPTION
**Pull request by ybykov-a9s**
Cover caching logic with unit and integration tests
Update golang and packaging utils versions
Update github actions

### CODE REVIEW

**Comment by ybykov-a9s** [comment]
Caching logic saves up to 30s of build time on a simple test application and more than 3m on a real project.
Caching logic is enabled if major and minor versions of ruby weren't changed, and if Gemfile and Gemfile.lock files weren't changed.

File: bundler/ruby_version_resolver.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+	getRubyVersionCmd := strings.Join([]string{
+		"ruby",
+		"--version",
+	}, " ")
+	cmdStdOut, err := RunBashCmd(getRubyVersionCmd, workingDir)
+	if err != nil {
+		return "", fmt.Errorf("failed to obtain ruby version: %w: %s", err, cmdStdOut)
+	}
+
+	versions := regexp.MustCompile(`ruby (\d+\.\d+\.\d+)`).FindStringSubmatch(cmdStdOut)
```
Comment (khaledavarteq) [inline bundler/ruby_version_resolver.go:32]: this RegEx matches versions like `1.9.3` but not versions like `1.9.3-p551` or versions like `jruby-9.2.17.0` which we should also support

**Review by khaledavarteq — CHANGES_REQUESTED** [review]

File: bundler/ruby_version_resolver.go (hunk 1 above)
Comment (ybykov-a9s) [reply bundler/ruby_version_resolver.go]: RVM is now a source for installed ruby version. Next regexpes are now used for detecting a ruby version:
		`(jruby-\d+\.\d+\.\d+).\d+`,
		`(jruby-\d+\.\d+)\.\d+`,
		`(jruby-head)`,
		`(ruby-\d+\.\d+)\.\d+`,
		`(ruby-head)`
So it covers the RVM's versions format for jruby and ruby: jruby-1.2.3.4, jruby-1.2.3, ruby-1.2.3 and `head` versions. Also all possible suffixes (aka patchlevel and prerelease) are supported but ignored.

File: bundler/ruby_version_resolver.go [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+// NewRubyVersionResolver initializes an instance of RubyVersionResolver.
+func NewRubyVersionResolver() RubyVersionResolver {
+	return RubyVersionResolver{}
+}
+
+// Lookup returns the version of Ruby installed in the build environment.
+func (r RubyVersionResolver) Lookup(workingDir string) (string, error) {
+	getRubyVersionCmd := strings.Join([]string{
+		"rvm",
```
Comment (khaledavarteq) [inline bundler/ruby_version_resolver.go:22]: the RVM Buildpack uses a different logic to determine the Ruby version. Do you see a potential risk coming from the fact that RVM and RVM Bundler use different Ruby version determination logics?

**Review by khaledavarteq — APPROVED** [review]

**Pull request merged**